modules/cas_cache/layer_cache_management.c

```diff
@@ -2353,12 +2353,27 @@ int cache_mngt_activate(struct ocf_mngt_cache_standby_activate_config *cfg,
 	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
 		goto out_cache_put;
 
-	cas_lazy_thread_stop(context->rollback_thread);
+	result = _cache_start_finalize(cache, -1, true);
+	if (result)
+        goto err;
 
+    cas_lazy_thread_stop(context->rollback_thread);
 	if (result)
 		goto err_free_context;
 
-	result = _cache_start_finalize(cache, -1, true);
+err:
+    _cache_mngt_async_context_init(&context->async);
```

Comment:
I'm not sure if calling `_cache_mngt_async_context_init()` for the second time is a good idea as it calls `init_completion()` which is not a correct usages of completions. According to [the documentation ](https://www.kernel.org/doc/Documentation/scheduler/completion.txt) you are supposed call `reinit_completion()`.